### PR TITLE
Add role for the GPTE Decision Manager 7 Foundations course

### PR DESCRIPTION
**Pull request by btison**
##### SUMMARY

Role for the GPTE Decision Manager 7 Foundations course

The current DM7 Foundations Course ELAB runs on OpenShift 3.11 on Novello. This new role moves the workload to a shared OCP 4.x cluster. 

##### ISSUE TYPE

- New role Pull Request

##### COMPONENT NAME

role name: ocp4_workload_dm7_foundations_elt

##### ADDITIONAL INFORMATION

Role for the GPTE Decision Manager 7 Foundations ELT course
Targets shared OCP 4.4 cluster
Deploys the following components:
* DM Business central v 7.8
* DM KIE Server v7.8
* Angular Front End application

All components are verified to be up and running by verifying the number of readyReplicas in the deployment status.

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/tasks/post_workload.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    namespace: openshift-console
+    name: console
+  register: r_console_route
+
+- name: Set webconsole address
+  set_fact:
+    webconsole: "https://{{ r_console_route.resources[0].spec.host }}"
+
+- name: Get API for command line
+  command: oc whoami --show-server
```
Comment (newgoliath) [inline ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/tasks/post_workload.yml:18]: We prefer this code now - no shell out.

  - name: Determine API server URL
    k8s_info:
      api_version: config.openshift.io/v1
      kind: Infrastructure
      name: cluster
    register: r_api_url

File: ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/files/image-pull-secret.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: registry-credentials
```
Comment (newgoliath) [inline ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/files/image-pull-secret.yml:5]: Remove creds.  Put in a local secret.

File: ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/templates/clusterresourcequota.j2 [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: quota.openshift.io/v1
+kind: ClusterResourceQuota
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/templates/clusterresourcequota.j2:3]: Is this workload intended to run on a shared cluster? If so, we already have a clusterresourcequota for users when they order access. How is this working with that?

File: ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/files/image-pull-secret.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
...U3kxdk1rNW1RblozTTJ4MU1GOXNPVlkyY2psbFUwWlRjM1JWTm5abE9HbDNjVkkyVFU5eGNscG5NSFkwYVROSk5WSjJOMDlaV0VaNFVFTjBSMWRxV21Wa00xRnlXVlZYY0VzNVRqWklNRmREWTE5T0xWWlZTMWhtUzJ4WlNpMTZaSFV3Vm1nMlFuRjRZbGhEYVV4SWEyVm5TR0ptYW1Gd1RrbEZiRlJpZWpsWlZtNTRMVU5qTVhwUE9EUlNRMWRMY2t0TVdrcG9XRVp1ZVVJeE1tcFpaRlJ2U0RkVlRXaDNlbEZTTTA1eGIwcFljMDV1Y0doNFQwcEpZMEpxTldWeGNGbG9WSFF4V0dSMlgzbHVYM0pyTmpWd2RWVm9VMG94YzBWZldHaEpkbUoxUzJsZlpITk5NRWhHY3poUlIxa3pOVXhtWTBKbWQyaHFkazlOVTJweGFGcGZSMjVtWXpkS1FuVkJXRU4wVnpoTFNWOUVVbVY2Wm1KbFNqRTNWRWRYVG5GeVRFRlBVRkU0TUhoRFdXWXRhMDVoVVZGMFlXNWFlalpFY0V4QldqWkhTSFF0YUMwNFQwTjBTRUprY1dScFdEUnVja0pCYTJaS2FFWTJSbW80UXpGaFQzbFBiMkpuUjIxc2EweHBjRzVyVlVSUGFYSTNjVmszT0VGMU56UXdVMmhTU0ZKbmJUTlVXbEpKZUVkV2JGTm9SMU5ZVkRGNk0yUkRWRVJqZUUxRVNtZHBORjlwVmxWNVlWbFVjV1pwTFVNNGJtdEZSalZPV1VkSFZuSnlURlpTVTNNd1dtdENNbFpIVlZZMVZVNVhWMVZTVVZSck5IaFpkbmgwV25GblRVNWlVRTQyTW14d01FSkZlQzE0V2pVMlJWSlVkVXhNTTJveVduZzRNVGxzYjJKQmNuVk9OMFJTU1c1b1EzTnVSRVYzWTJKdFEyeFpSVWh5UVU5VFNrbzVjVzV0YzJnNFpISmZTVEUzWjFreFgxZDBlR3AxTlZsRFdUbFlWVWRUV0ZOelZtNDBiR0p5U1VoaE9VWllTVXhHU2pGRU9IbHBSREI1YnpkTWQzZHNTVTFqVldWTFlsaDVhMkZHUlZRMVFsWXRlbWxYT1VKR09VZEdOVWRHT0Y5blgyeE9kMUZJWWxaelJVdG5abHA1VGxGQ2VrcExNbkZxTFZkU1pUQnIifX19
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/files/image-pull-secret.yml:7]: This needs to be removed from here.

File: ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/tasks/workload_dm.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
+---
+
+- name: create image pull secret for registry.redhat.io
+  k8s:
+    state: present
+    namespace: "{{ ocp4_workload_dm7_foundations_elt_project_dm }}"
+    definition: "{{ lookup('file', role_path ~ '/files/image-pull-secret.yml' ) | from_yaml }}"
+
+- name: link secret to default sa
+  shell: oc secrets link default registry-credentials --for=pull -n {{ ocp4_workload_dm7_foundations_elt_project_dm }}
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/tasks/workload_dm.yml:10]: Unless you need shell functions or variables, `command` module is typically preferred.

**Pull request merged**